File: topics/procedure-decl/examples/MorseCalling.pas
// Program: MorseCalling - GameMain.pas
// Outputs audio for the CQ 'Calling Anyone' signal.
program MorseCalling;

procedure ShortSignal();
begin
    PlaySoundEffect('dit');
    Delay(200);
end;

procedure LongSignal();
begin
    PlaySoundEffect('dah');
    Delay(600);
end;

procedure SignalC();
begin
    Delay(600);
    LongSignal();
    ShortSignal();
    LongSignal();
    ShortSignal();
end;

procedure SignalQ();
begin
    Delay(600);
    LongSignal();
    LongSignal();
    ShortSignal();
    LongSignal();
end;

procedure LoadSounds();
begin
    LoadSoundEffectNamed('dah', 'dah.wav');
    LoadSoundEffectNamed('dit', 'dit.wav');
end;

procedure Main();
begin
    OpenAudio();
    LoadSounds();
    
    SignalC();
    SignalQ();
    
    CloseAudio();
    ReleaseAllResources();
end;

begin
    Main();
end.

File: topics/procedure-decl/examples/morse-calling.c
/* Program: MorseCalling - main.c
*  Outputs audio for the CQ "Calling Anyone" signal.
*/
#include <stdio.h>
#include "SwinGame.h"

void short_signal()
{
    play_sound_effect("dit");
    delay(200);
}

void long_signal()
{
    play_sound_effect("dah");
    delay(600);
}

void signal_c()
{
    delay(600);
    long_signal();
    short_signal();
    long_signal();
    short_signal();
}

void signal_q()
{
    delay(600);
    long_signal();
    long_signal();
    short_signal();
    long_signal();
}

void load_sounds()
{
    load_sound_effect_named("dah", "dah.wav");
    load_sound_effect_named("dit", "dit.wav");
}

int main()
{
    open_audio();
    
    load_sounds();
    
    signal_c();
    signal_q();
    
    close_audio();
    release_all_resources();
    return 0;
}